mobx_codegen/lib/src/type_names.dart

```diff
@@ -75,20 +73,12 @@ class LibraryScopedNameFinder {
     return _getTypeName(typeParameter.bound);
   }
 
-  /// Calculates a type name, including its type arguments if
-  /// [includeTypeArguments] is `true`.
+  /// Calculates a type name, including its type arguments
   ///
   /// The returned string will include import prefixes on all applicable types.
-  String _getTypeName(DartType type, {bool includeTypeArguments = false}) {
+  String _getTypeName(DartType type) {
     final sb = StringBuffer()..write(_findUnparameterizedTypeName(type));
 
-    if (type is ParameterizedType && includeTypeArguments) {
```

Comment:
This was causing issues as it would loop through type arguments and generate an invalid type: such as `List<User><User>`